### DISCUSSION TITLE
Batch setup

### DISCUSSION THREAD

**Discussion by mtanasiewicz** [RavenSystem/esp-homekit-devices#2101] [Q&A]
Hello!

At first thank you for excellent work!

I have a problem to solve. Recently I have upgraded almost a hundred devices from really old firmware to the latest one. Now I need to update all those devices with new json configurations. I also want to update all of them to seek for the strongest AP signal from WIFI Normal mode. Doing it with Haa Manager is unfortunately quite tedious because I have to do it one by one: setting device to setup mode -> updating configuration -> saving. I could'nt find a batch setup update in HAA manager. Some of the devices in our network have weak signal and reconfiguring it the default way works preety slow by UI, I have to do it with POSTMAN. This operation takes ages for almost hundred devices...

I figured out a POST request that is made to store the new configuration and I came up with an idea that I could write a simple script to batch update configurations by ip addresses. However to do that I need to somehow put all those devices into setup mode before storing the configuration, the device won't let me otherwise. Could you please tell me what request should I issue to put device into setup mode? I tried to search the repositories and discussions to find answers, bu couldn't find one. Maybe there is some API documentation that I'm not aware?

All the best!

**Comment by RavenSystem** [comment]
You are right about POST request to send new config when device is in setup mode. But in normal mode, there is not any hidden service or API to put it in setup mode. In normal mode, only HomeKit service and its protocol (HAP) are used, for security reasons. This means that you must use HAP to put device in setup mode. I personally don't know any kind of HomeKit CLI or similar to do a batch process to put devices in setup mode.

**Reply by mtanasiewicz** [reply]
Thank you for detailed answer.
